lib/prism.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+const Prism = require('prismjs');
+const prismLoadLanguages = require('prismjs/components/');
+
+// https://github.com/PrismJS/prism/issues/2145
+const prismComponents = require('prismjs/components');
+const prismSupportedLanguages = Object.keys(prismComponents.languages);
+
+const escapeHTML = require('./escape_html');
+const unescapeHTML = require('./unescape_html');
+
+/**
+ * Wrapper of Prism.highlight()
+ * @param {String} code
+ * @param {String} language
+ */
+function prismHighlight(code, language) {
+  console.log(Object.keys(Prism.languages));
+  // Prism has not load the language pattern
+  if (!Prism.languages[language] && prismSupportedLanguages.includes(language)) prismLoadLanguages(language);
+
+  if (Prism.languages[language]) {
+    // Prism escapes output by default
+    return Prism.highlight(unescapeHTML(code), Prism.languages[language], language);
+  }
+
+  // Current language is not supported by Prism, return origin code;
+  return escapeHTML(code);
+}
+
+/**
+ * Generate line number required HTML snippet
+ * @param {String} code - Highlighted code
+ */
+function lineNumberUtil(code) {
+  const matched = code.match(/\n(?!$)/g);
+  const num = matched ? matched.length + 1 : 1;
+  const lines = new Array(num + 1);
+
+  return `<span aria-hidden="true" class="line-numbers-rows">${lines.join('<span></span>')}</span>`;
+}
+
+function replaceTabs(str, tab) {
+  return str.replace(/^\t+/gm, match => {
+    let result = '';
+
+    for (let i = 0, len = match.length; i < len; i++) {
+      result += tab;
+    }
+
+    return result;
+  });
+}
+
+function PrismUtil(str, options = {}) {
+  if (typeof str !== 'string') throw new TypeError('str must be a string!');
+
+  const {
+    lineNumber = true,
+    language = 'plain',
+    tab
+  } = options;
+
```

Comment:
Is is necessary to translate the language name `plain` to `none` ?

In Hexo (with `highlightjs`), the language name `plain` means "no highlighting".
By the document of Prism, it is achieved by using the class name `language-none`.
https://prismjs.com/#basic-usage

I think the manner in Hexo should keep regardless what highlighter is used.
So above translation will be desirable.

Reply:
I will update to that.

Reply:
I have updated the default `lang` to `none` to be consistent with `prismjs`.

`plain` is still acceptable, while `{ lang: 'plain' }` will result in `.language-none`.

Reply:
Thanks. I recognized!